spec.html

```diff
@@ -12677,51 +12615,46 @@ <h1>
-            1. If _current_.[[Configurable]] is *false* and _current_.[[Writable]] is *false*, then
+            1. If ! IsGenericDescriptor(_Desc_) is *false* and ! SameValue(IsAccessorDescriptor(_Desc_), IsAccessorDescriptor(_current_)) is *false*, return *false*.
+            1. If ! IsAccessorDescriptor(_Desc_) is *true*, then
+              1. If _Desc_.[[Get]] is present and ! SameValue(_Desc_.[[Get]], _current_.[[Get]]) is *false*, return *false*.
+              1. If _Desc_.[[Set]] is present and ! SameValue(_Desc_.[[Set]], _current_.[[Set]]) is *false*, return *false*.
+            1. Else if _current_.[[Writable]] is *false*, then
               1. If _Desc_.[[Writable]] is present and _Desc_.[[Writable]] is *true*, return *false*.
-              1. If _Desc_.[[Value]] is present and SameValue(_Desc_.[[Value]], _current_.[[Value]]) is *false*, return *false*.
-              1. Return *true*.
-          1. Else,
-            1. Assert: ! IsAccessorDescriptor(_current_) and ! IsAccessorDescriptor(_Desc_) are both *true*.
-            1. If _current_.[[Configurable]] is *false*, then
-              1. If _Desc_.[[Set]] is present and SameValue(_Desc_.[[Set]], _current_.[[Set]]) is *false*, return *false*.
-              1. If _Desc_.[[Get]] is present and SameValue(_Desc_.[[Get]], _current_.[[Get]]) is *false*, return *false*.
-              1. Return *true*.
+              1. If _Desc_.[[Value]] is present and ! SameValue(_Desc_.[[Value]], _current_.[[Value]]) is *false*, return *false*.
           1. If _O_ is not *undefined*, then
-            1. For each field of _Desc_ that is present, set the corresponding attribute of the property named _P_ of object _O_ to the value of the field.
+            1. If ! IsDataDescriptor(_current_) is *true* and ! IsAccessorDescriptor(_Desc_) is *true*, then
+              1. If _Desc_ has a [[Configurable]] field, let _configurable_ be Desc.[[Configurable]], else let _configurable_ be _current_.[[Configurable]].
+              1. If _Desc_ has a [[Enumerable]] field, let _enumerable_ be Desc.[[Enumerable]], else let _enumerable_ be _current_.[[Enumerable]].
+              1. Replace the property named _P_ of object _O_ with an accessor property having [[Configurable]] and [[Enumerable]] attributes to _configurable_ and _enumerable_, respectively, and each other attribute set to its corresponding value in _Desc_ if present, otherwise to its <emu-xref href="#table-object-property-attributes">default value</emu-xref>.
+            1. Else if ! IsAccessorDescriptor(_current_) is *true* and ! IsDataDescriptor(_Desc_) is *true*, then
+              1. If _Desc_ has a [[Configurable]] field, let _configurable_ be Desc.[[Configurable]], else let _configurable_ be _current_.[[Configurable]].
+              1. If _Desc_ has a [[Enumerable]] field, let _enumerable_ be Desc.[[Enumerable]], else let _enumerable_ be _current_.[[Enumerable]].
```

Comment:
I'd still prefer to avoid this duplication, but it works and is an improvement over the current spec.